### PR TITLE
[ENH] Add LSTM-FCN estimator from sktime-dl

### PR DESCRIPTION
**Pull request by solen0id**
#### What does this implement/fix? Explain your changes.
Add LSTM-FCN estimator from sktime-dl

I am using sktime & sktime-dl in university as part of a course project this semester. I would love to use the LSTM-FCN model from sktime-dl in my project, but since sktime-dl comes with some pretty old (and hard-pinned) dependencies, I decided to try and port the model over to sktime.

I made some very small changes to parameter names to stay in line with the existing models, such as the CNNClassifier (e.g changed `nb_epochs` to `n_epochs`).

Added jupyter notebook `02b_classification_multivariate_lstmfcn.ipynb` with a small demo based on the CNN example.

Ran `pytest -k LSTMFCNClassifier` without errors.

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### What should a reviewer concentrate their feedback on?
Please focus on everything, except the network architecture itself. I did not implement it, all credit goes to Jack Russon from sktime-dl. [Source](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/networks/_lstmfcn.py). 

I am happy to incorporate any other feedback you might have!

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [x] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by fkiraly** [comment]
Thanks!

Your linting is failing, kindly check code formatting, here is the guide:
https://www.sktime.org/en/stable/developer_guide/coding_standards.html

**Comment by solen0id** [comment]
Hey @fkiraly, thanks for the initial review! 
I ran the pre-commit suite a bunch of times and fixed up all errors that popped up, let me know what you think.

File: sktime/networks/lstmfcn.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+"""LongShort Term Memory Fully Convolutional Network (LSTM-FCN)."""
+
+__author__ = "Jack Russon"
```
Comment (fkiraly) [inline sktime/networks/lstmfcn.py:4]: please use your GitHub ID

File: sktime/classification/deep_learning/lstmfcn.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    Ported from sktime-dl source code
+    https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_lstmfcn.py
+
+    References
+    ----------
+    .. [1] Karim et al. Multivariate LSTM-FCNs for Time Series Classification, 2019
+    https://arxiv.org/pdf/1801.04503.pdf
+
+    """
+
```
Comment (fkiraly) [inline sktime/classification/deep_learning/lstmfcn.py:67]: you need to set the `python_dependencies` tag so the soft dependency `tensorflow` is properly isolated - have a look how the other neural networks do this

**Review by fkiraly — CHANGES_REQUESTED** [review]
Looks overall reasonable!

The remaining issue is to isolate the soft dependency tensorflow. Please use the tag as stated above (in both classes), and all should be fine!

File: sktime/networks/lstmfcn.py (hunk 1 above)
Comment (solen0id) [reply sktime/networks/lstmfcn.py:4]: Jack Russon is the person who originally implemented the network over in sktime-dl.

Is it alright if I extend the authors to `__author__ = ["Jack Russon", "solen0id"]`? It wouldn't feel right to just put my name on his work

File: sktime/classification/deep_learning/lstmfcn.py (hunk 2 above)
Comment (solen0id) [reply sktime/classification/deep_learning/lstmfcn.py:67]: Done, added the `python_dependencies` tag here in the estimator and in the corresponding network

**Comment by solen0id** [comment]
All done, let me know if the suggested author attribution works for you

**Comment by solen0id** [comment]
I replaced the actual grid search in the example notebook with a comment about grid search compatibility to speed up things

File: sktime/networks/lstmfcn.py (hunk 1 above)
Comment (fkiraly) [reply sktime/networks/lstmfcn.py]: oh, apologies, I thought you were Jack Russon.

In that case, replace Jack Russon by his GitHub ID `jnrusson1` (and check whether he is in the contributor file)

Comment (solen0id) [reply sktime/networks/lstmfcn.py]: All done, added `jnrusson1` and myself to the `.all-contributorsrc` file :+1:

**Comment by solen0id** [comment]
Hey @fkiraly, I added Jack and myself to the contributors file.  Let me know if there is anything else I should fix up :)

**Comment by solen0id** [comment]
I moved the `AttentionLSTM` class into the `networks` package, that should fix the failing soft dependency test :crossed_fingers:

**Comment by solen0id** [comment]
Hey @fkiraly, I merged the latest changes from upstream to resolve the merge conflict in `.all_contributorsrc` that just appeared. Would you kindly authorise another execution of the test suite please? Thanks 🙏

**Comment by solen0id** [comment]
Hey @fkiraly looks like there is a test failing, although it seems unrelated to any of my changes. Do you have an idea whats going on here?


```
FAILED sktime/classification/tests/test_base.py::test_deep_estimator_full[keras-adamax] - ValueError: `optimizer` of type <class 'keras.optimizers.optimizer_v2.adamax.Adamax'> cannot be serialized, it should either be absent/None/str/tf.keras.optimizers.Optimizer object
```

**Comment by solen0id** [comment]
I pulled in your changes for #3817, that should hopefully do it now 🤞

**Comment by solen0id** [comment]
Can we do another test run @fkiraly?

**Comment by fkiraly** [comment]
Of course! Sorry for being silent on this. Thanks for your contribution!

**Comment by solen0id** [comment]
All good, no need to apologize :)

I see there are still some OS + Python version combinations causing errors in the test matrix, I'll try to figure out what's wrong there

**Review by fkiraly — CHANGES_REQUESTED** [review]
The tests failing are the ones commonly failing due to incomplete random seed setting in `keras`. This is a known issue and will be dealt with for all `keras` estimators simultaneously, if it will be dealt with.

So, you should add the tests as exceptions to `tests._config`, similar to the other `keras` based estimators.

**Comment by solen0id** [comment]
Good stuff, thanks for the hint :+1:

**Comment by solen0id** [comment]
Shall we give the tests another go @fkiraly ? :)

**Comment by solen0id** [comment]
Looks like a compiler error occured during install of dependencies on macOS, looks to me like it might have just been a fluke. Do you agree @fkiraly? Maybe we can just re-run the test suite, everything else looks fine now

**Comment by solen0id** [comment]
Hey @fkiraly, could you please authorize another run of the test suite? Thank you!

**Comment by fkiraly** [comment]
oops, sorry - did miss this one. Of course, let's see

**Pull request merged**